File: Easy/14. Longest Common Prefix/14. Longest Common Prefix.ts
function longestCommonPrefix(strs: string[]): string {
    if (strs === null || strs.length === 0) {
        return '';
    }

    const firstStr: string = strs[0];
    const prefixLen: number = firstStr.length;

    for (let i = 0; i < prefixLen; i++) {
        const c: string = firstStr.charAt(i);
        for (let j = 1; j < strs.length; j++) {
            const str: string = strs[j];
            if (i >= str.length || str.charAt(i) !== c) {
                return firstStr.substring(0, i);
            }
        }
    }

    return firstStr;
}

File: Easy/14. Longest Common Prefix/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        
        String firstStr = strs[0];
        int prefixLen = firstStr.length();
        
        for (int i = 0; i < prefixLen; i++) {
            char c = firstStr.charAt(i);
            for (int j = 1; j < strs.length; j++) {
                String str = strs[j];
                if (i >= str.length() || str.charAt(i) != c) {
                    return firstStr.substring(0, i);
                }
            }
        }
        
        return firstStr;
    }
}